churaverse-plugins-client/src/churarenPlugin/churarenAlchemyItems/iceArrowPlugin/iceArrowPlugin.ts

```diff
@@ -0,0 +1,172 @@
+  }
+
+  protected useAlchemyItem = (ev: UseAlchemyItemEvent): void => {
+    if (ev.alchemyItem.kind !== 'iceArrow') return
+    const gap = 65
+    for (let i = 0; i < 4; i++) {
+      const renderer = this.iceArrowPluginStore.iceArrowAttackRendererFactory.build()
+      if (renderer === undefined) return
+      const startPos = ev.ownPlayer.position.copy()
+      // 4方向に分散するための角度を計算 (ラジアン)
+      const angle = -(i * Math.PI) / 2 // 90度 (π/2) ごとに回転
+
+      // x, y成分で方向ベクトルを計算
+      const attackVector = {
+        x: parseFloat(Math.cos(angle).toFixed(2)),
+        y: parseFloat(Math.sin(angle).toFixed(2)),
+      }
+      const position = new Position(startPos.x + gap * attackVector.x, startPos.y + gap * attackVector.y)
+      const iceArrow = new IceArrow(
+        uniqueId(),
+        ev.ownPlayer.id,
+        position,
+        ev.ownPlayer.direction,
+        Date.now(),
+        attackVector
+      )
+      this.iceArrowPluginStore.iceArrows.set(iceArrow.iceArrowId, iceArrow)
+      this.iceArrowPluginStore.iceArrowAttackRenderers.set(iceArrow.iceArrowId, renderer)
+      if (iceArrow.ownerId === this.playerPluginStore.ownPlayerId) {
+        const iceArrowSpawnMessage = new IceArrowSpawnMessage({
+          iceArrowId: iceArrow.iceArrowId,
+          startPos: iceArrow.position.toVector() as Vector & Sendable,
+          direction: iceArrow.direction,
+          spawnTime: iceArrow.spawnTime,
+          attackVector: iceArrow.attackVector as Vector & Sendable,
+        })
+        this.networkStore.messageSender.send(iceArrowSpawnMessage)
+      }
+      this.walkIceArrow(iceArrow, renderer)
+    }
+    const clearAlchemyItemBoxEvent = new ClearAlchemyItemBoxEvent(ev.ownPlayer.id)
+    this.bus.post(clearAlchemyItemBoxEvent)
+  }
+
+  private readonly spawnIceArrow = (ev: EntitySpawnEvent): void => {
+    if (!(ev.entity instanceof IceArrow)) return
+    const iceArrow = ev.entity
+    const renderer = this.iceArrowPluginStore.iceArrowAttackRendererFactory.build()
+    if (renderer === undefined) return
+    this.iceArrowPluginStore.iceArrows.set(iceArrow.iceArrowId, iceArrow)
+    this.iceArrowPluginStore.iceArrowAttackRenderers.set(iceArrow.iceArrowId, renderer)
+    this.walkIceArrow(iceArrow, renderer)
+  }
+
+  private readonly dieIceArrow = (ev: EntityDespawnEvent): void => {
+    if (!(ev.entity instanceof IceArrow)) return
+    const iceArrowId = ev.entity.iceArrowId
+    const iceArrow = this.iceArrowPluginStore.iceArrows.get(iceArrowId)
+    const iceArrowRenderer = this.iceArrowPluginStore.iceArrowAttackRenderers.get(iceArrowId)
+    iceArrow?.die()
+    iceArrowRenderer?.dead()
+    this.iceArrowPluginStore.iceArrows.delete(iceArrowId)
+    this.iceArrowPluginStore.iceArrowAttackRenderers.delete(iceArrowId)
+  }
+
+  private walkIceArrow(iceArrow: IceArrow, renderer: IIceArrowAttackRenderer): void {
+    // 斜め方向も水平・垂直方向も全て同じ移動距離にする
+    const distance = 1.0 * ICE_ARROW_WALK_LIMIT_GRIDS * GRID_SIZE
```

Comment:
型的には、同じnumberのままだと思うので1.0を掛ける必要はなさそうです

```suggestion
    const distance = ICE_ARROW_WALK_LIMIT_GRIDS * GRID_SIZE
```